### PR TITLE
fix: handle leapyear in RFC3339.toPosix correctly

### PR DESCRIPTION
**Pull request by tufteddeer**
Detecting whether the current year is a leapyear was wrong, so dates in leapyears after february where one day off

### CODE REVIEW

File: tests/RFC3339Test.elm [hunk 1]
```diff
@@ -345,6 +345,48 @@ decodeTimestampTest =
...
+                , 2172
+                , 1976
+                , 2076
+                , 2176
+                , 2092
+                , 2192
+                , 1996
+                , 2096
+                , 2196
+                ]
```
Comment (Flo0807) [inline tests/RFC3339Test.elm]: Do we want to add some end-of-century years to the tests?

```suggestion
                , 2096
                , 2196
                , 4000
                , 8000
                ]
```

File: tests/RFC3339Test.elm [hunk 2]
```diff
@@ -345,6 +345,48 @@ decodeTimestampTest =
...
+                    |> List.all RFC3339.isLeapyear
+                    |> Expect.equal True
+            )
+        , test "with normal years"
+            (\_ ->
+                [ 2021, 2022, 2015, 1905, 2195 ]
```
Comment (Flo0807) [inline tests/RFC3339Test.elm]:
```suggestion
                [ 2000, 2021, 2022, 2015, 1905, 2195 ]
```

**Review by Flo0807 — APPROVED** [review]
Good catch! 🚀

File: tests/RFC3339Test.elm (hunk 1 above)
Comment (tufteddeer) [reply tests/RFC3339Test.elm]: Good idea, done!

**Pull request merged**